index.js

```diff
@@ -61,12 +61,36 @@ class RpcResponseError {
   }
 }
 
-exports.load = function(host, serviceName, options) {
-  const metaPath = getMetaPath(serviceName);
-  const client = new Client(host, options);
+class Registry {
+  constructor() {
+    this._registeredServices = {};
+  }
+
+  load(host, serviceName, options) {
+    const metaPath = getMetaPath(serviceName);
+    const client = new Client(host, options);
+
+    const loadClient = client.load(metaPath);
+    this._registeredServices[serviceName] = loadClient;
+
+    return loadClient;
+  }
+
+  createWellKnownHandler() {
+    return (req, res) => {
+      const services = Object.keys(this._registeredServices).map((s) => {
+        return { name: s };
+      });
+
+      res.json({
+        services,
+      });
+    };
+  }
+}
 
-  return client.load(metaPath);
-};
+exports.registry = new Registry();
+exports.Registry = Registry;
```

Comment:
Similar to the other PR, this should export `WELL_KNOWN_META_PATH` which will be `/client` in this case, not `/server`